sidecar/src/self_telemetry.rs

```diff
@@ -60,62 +64,93 @@ pub fn self_telemetry(
-                                metrics.collect_and_send().await;
-                                let _ = worker.send_msg(TelemetryActions::Lifecycle(LifecycleAction::FlushMetricAggr)).await;
-                                let _ = worker.send_msg(TelemetryActions::Lifecycle(LifecycleAction::FlushData)).await;
-                            },
-                            _ = shutdown_receiver.recv() => {
-                                metrics.collect_and_send().await;
-                                let _ = worker.send_msg(TelemetryActions::Lifecycle(LifecycleAction::Stop)).await;
-                                let _ = join_handle.await;
-                                return
-                            },
-                        }
-                    }
-                } else {
-                    shutdown_receiver.recv().await;
-                }
+                let worker_cfg = SelfTelemetry { submission_interval, watchdog_handle, config, server };
+                worker_cfg.spawn_worker().await
             },
         }
     })
 }
+
+pub struct SelfTelemetry {
+    pub submission_interval: tokio::time::Interval,
+    pub watchdog_handle: WatchdogHandle,
+    pub config: ddtelemetry::config::Config,
+    pub server: SidecarServer,
+}
+
+impl SelfTelemetry {
+    /// spawn_worker
+    ///
+    /// should always succeed
+    /// not to bring down other functionality if we fail to initialize the internal telemetry
+    pub async fn spawn_worker(mut self) {
+        let (worker, join_handle) = match TelemetryWorkerBuilder::new_fetch_host(
+            "datadog-ipc-helper".to_string(),
+            "php".to_string(),
+            "SIDECAR".to_string(),
+            env!("CARGO_PKG_VERSION").to_string(),
+        )
+        .spawn_with_config(self.config.clone())
+        .await
+        {
+            Ok(r) => r,
+            Err(_err) => {
+                self.watchdog_handle.wait_for_shutdown().await;
+                return;
+            }
+        };
+
+        let metrics = MetricData {
+            worker: &worker,
+            server: &self.server,
+            sidecar_watchdog: &self.watchdog_handle,
+            submitted_payloads: worker.register_metric_context(
+                "server.submitted_payloads".to_string(),
+                vec![],
+                MetricType::Count,
+                true,
+                MetricNamespace::Sidecar,
+            ),
+            active_sessions: worker.register_metric_context(
+                "server.active_sessions".to_string(),
+                vec![],
+                MetricType::Gauge,
+                true,
+                MetricNamespace::Sidecar,
+            ),
+            memory_usage: worker.register_metric_context(
+                "server.memory_usage".to_string(),
+                vec![],
+                MetricType::Gauge,
+                true, // TODO: make sure its a common metric
+                MetricNamespace::Sidecar,
+            ),
```

Comment:
This is not going to do what you expect. We don't tag telemetry metrics by runtime-id, only by the cardinality of a service.
So your gauges are not going to be aggregated correctly. If we receive multiple points, only one of them is going to stay.

What you want is a distribution

Reply:
What is a distribution / how do I use that? Under MetricType I just see Gauge and Count as options.
Or are you talking about something that doesn't exist yet?